src/client/Microsoft.Identity.Client/TelemetryCore/Internal/HttpTelemetryContent.cs

```diff
@@ -18,49 +20,81 @@ public HttpTelemetryContent(EventBase evt)
                 evt.TryGetValue(MsalTelemetryBlobEventNames.ApiErrorCodeConstStrKey, out string errorCode);
                 evt.TryGetValue(MsalTelemetryBlobEventNames.ForceRefreshId, out string forceRefresh);
 
-                ApiId = apiId ?? string.Empty;
-                CorrelationId = correlationId ?? string.Empty;
-                LastErrorCode = errorCode ?? string.Empty;
+                ApiId.Add(apiId ?? string.Empty);
+                CorrelationId.Add(correlationId ?? string.Empty);
+                LastErrorCode.Add(errorCode ?? string.Empty);
                 ForceRefresh = forceRefresh ?? string.Empty;
             }
         }
 
-        public string LastErrorCode { get; set; } = string.Empty;
-        public string ApiId { get; set; } = string.Empty;
-        public string CorrelationId { get; set; } = string.Empty;
+        public /*For testing*/ HttpTelemetryContent(bool shouldClearCache)
+        {
+            if (shouldClearCache)
+            {
+                ApiId?.Clear();
+                CorrelationId?.Clear();
+                LastErrorCode?.Clear();
+            }
+        }
+
         public string ForceRefresh { get; set; } = string.Empty;
+        public List<string> ApiId { get; set; } = new List<string>();
+        public List<string> CorrelationId { get; set; } = new List<string>();
+        public List<string> LastErrorCode { get; set; } = new List<string>();
 
-        public string GetCsvAsPrevious()
+        public string GetCsvAsPrevious(int successfulSilentCallCount)
         {
-            if (string.IsNullOrWhiteSpace(ApiId))
+            if (ApiId == null)
             {
                 return string.Empty;
             }
 
             // csv expected format:
-            // 1|api_id,correlation_id,last_error_code|platform_config
-            string[] myValues = new string[] {
-                ApiId,
-                CorrelationId,
-                LastErrorCode};
+            // 2|silent_successful_count|failed_requests|errors|platform_fields
+            // failed_request can be further expanded to include:
+            // api_id_1,correlation_id_1,api_id_2,correlation_id_2|error_1,error_2
+            string apiIdCorIdData = CreateApiIdAndCorrelationIdContent();
+            string data = string.Format(CultureInfo.InvariantCulture,
+                $"{TelemetryConstants.HttpTelemetrySchemaVersion2}{TelemetryConstants.HttpTelemetryPipe}{successfulSilentCallCount}{TelemetryConstants.HttpTelemetryPipe}" +
+                $"{CreateFailedRequestsContent(apiIdCorIdData)}" +
+                $"{TelemetryConstants.HttpTelemetryPipe}");
 
-            string csvString = string.Join(",", myValues);
-            csvString = $"{TelemetryConstants.HttpTelemetrySchemaVersion1}{TelemetryConstants.HttpTelemetryPipe}{csvString}{TelemetryConstants.HttpTelemetryPipe}";
-            return csvString;
+            if (data.Length <= 3800)
```

Comment:
No magic numbers please. Please have a constant documenting this number. Also pls read https://www.joelonsoftware.com/2003/10/08/the-absolute-minimum-every-software-developer-absolutely-positively-must-know-about-unicode-and-character-sets-no-excuses/

Reply:
Pretty sure the chars in that header are 1 byte / char.

Reply:
C# char is 16bit by default?

Reply:
2 bytes.